admin/Makefile

```diff
@@ -1,80 +1,36 @@
-BUILD_DIR ?= .build
-VENV ?= ${BUILD_DIR}/venv
-DOCKER_TAG ?= latest
-
-GIT_HASH := $(shell git rev-parse HEAD)
-DOCKER_BASE = camptocamp/c2cgeoportal
+# Used for development purpose: please do not remove this file
 
-.PHONY: build
-build: ${BUILD_DIR}/requirements.timestamp ${BUILD_DIR}/node_modules.timestamp
-
-.PHONY: clean
-clean:
-	rm -rf ${BUILD_DIR}/*.timestamp
-
-.PHONY: cleanall
-cleanall:
-	rm -rf ${BUILD_DIR}
-	rm -rf node_modules
+DOCKER_BASE ?= camptocamp/c2cgeoportal
+DOCKER_TAG ?= latest
 
-.PHONY: test
-test: build ${BUILD_DIR}/requirements-dev.timestamp
-	# Run your project's tests.
-	${VENV}/bin/pytest --driver Chrome
+default: help
 
-.PHONY: check
-check: flake8
+.PHONY: help
+help: ## Display this help message
+	@echo "Usage: make <target>"
+	@echo
+	@echo "Possible targets:"
+	@grep -Eh '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "    %-20s%s\n", $$1, $$2}'
 
-.PHONY: flake8
-flake8: ${BUILD_DIR}/requirements-dev.timestamp
-	${VENV}/bin/flake8 c2cgeoportal_admin acceptance_tests *.py
+.PHONY: preparedev
+preparedev: ## Run this once before serve
+	make -C .. build-tools build-test-db
+	make docker-build-admin
+	docker-compose up -d db
+	docker-compose run --rm admin wait-db
+	docker-compose run --rm admin alembic --config=/opt/c2cgeoportal/commons/alembic.ini --name=main upgrade head
+	docker-compose run --rm admin alembic --config=/opt/c2cgeoportal/commons/alembic.ini --name=static upgrade head
 
 .PHONY: serve
-serve: build
-	# Run your project.
-	${VENV}/bin/pserve --reload development.ini
-
-${BUILD_DIR}/venv.timestamp:
-	# Create a Python virtual environment.
-	virtualenv -p python3 ${VENV}
-	# Upgrade packaging tools.
-	${VENV}/bin/pip install --upgrade pip==9.0.1 setuptools==36.5.0
-	touch $@
+serve: ## Run c2cgeoportal_admin in development mode
+serve:
+	docker-compose up admin
 
-${BUILD_DIR}/requirements.timestamp: \
-		${BUILD_DIR}/venv.timestamp \
-		requirements.txt \
-		../commons/setup.py
-	# Install the project in editable mode.
-	${VENV}/bin/pip install -r requirements.txt
-	${VENV}/bin/pip install -e ../commons
-	${VENV}/bin/pip install -e .
-	touch $@
-
-${BUILD_DIR}/requirements-dev.timestamp: \
-		${BUILD_DIR}/requirements.timestamp \
-		requirements-dev.txt
-	# Install the project in editable mode with its testing requirements.
-	${VENV}/bin/pip install -r requirements-dev.txt
-	touch $@
-
-package.json:
-	cp ../$@ $@
-
-${BUILD_DIR}/node_modules.timestamp: package.json
-	npm install
-	touch $@
-
-.PHONY: docker-build
-docker-build:
-	cd .. ; \
-	docker build -t $(DOCKER_BASE)_admin:$(DOCKER_TAG) --build-arg GIT_HASH=$(GIT_HASH) -f admin/Dockerfile .
-
-.PHONY: docker-serve
-docker-serve:
-	docker-compose up
+.PHONY: clean
+clean: ## Stop composition, remove container and images
+	docker-compose down --remove-orphans
+	docker rm $(DOCKER_BASE)_admin:$(DOCKER_TAG)
```

Comment:
Shouldn't be rmi? 
Personally I don't clean the images on a clean, should we generalize that?

Reply:
We could consider, I've done it recently in getitfixed. I found useful such tool to get free disk space without removing all docker images of all projects.

Reply:
ok, but it should be rmi instead of rm